cycledash/auth.py

```diff
@@ -107,3 +108,33 @@ def register():
             login_user(user)
             return redirect('/')
     return render_template('register.html', errors=errors)
+
+def check_login_from_key(api_key):
+    username, password = api_key.split(":")
+    return check_login(username, password)
+
+def load_user_from_request(request):
+    """Support for basic authorization."""
+    # first, try to login using the api_key url arg
```

Comment:
@ihodes: without this one, my requests within `helpers.ResourceTest` were not being authorized and being forwarded to the `login` page automatically. I realized that all the `api` pages work fine with `ResourceTest`'s `get` and `post` requests, but none of our pages with `@login_required` decorator. 

I implemented this out of my ignorance about a better way; happy to hear your thoughts on this.